x/intrarelayer/types/proposal.go

```diff
@@ -69,6 +70,10 @@ func (rtbp *RegisterCoinProposal) ValidateBasic() error {
 		return err
 	}
 
+	if err := ibctransfertypes.ValidateIBCDenom(rtbp.Metadata.Base); err != nil {
```

Comment:
nice, we should also check that the metadata has the correct format

Reply:
following up on this, we should check that the format for IBC vouchers is the same as the one you wrote on the tests